internal/staging/stdscript/README.md

```diff
@@ -0,0 +1,218 @@
+stake generation p2pkh    |    Y   | `[]byte`
+stake generation p2sh     |    Y   | `[]byte`
+stake revocation p2pkh    |    Y   | `[]byte`
+stake revocation p2sh     |    Y   | `[]byte`
+stake change p2pkh        |    Y   | `[]byte`
+stake change p2sh         |    Y   | `[]byte`
+treasury add              |    Y   | -
+treasury generation p2pkh |    Y   | `[]byte`
+treasury generation p2sh  |    Y   | `[]byte`
+
+Abbreviations:
+
+* p2pk = Pay-to-public-key
+* p2pkh = Pay-to-public-key-hash
+* p2sh = Pay-to-script-hash
+
+## Package Usage Primer
+
+Interacting with existing standard scripts typically falls into the following
+categories:
+
+- Analyzing them to determine their type
+- Extracting data
+
+### Determining Script Type
+
+In order to provide a more ergonomic and efficient API depending on the specific
+needs of callers, this package offers three approaches to determining script
+types:
+
+1. Determining the script type as a `ScriptType` enum.  For example,
+   `DetermineScriptType` and `DetermineScriptTypeV0`.
+2. Determining if a script is a specific type directly.  For example,
+   `IsPubKeyHashScript`, `IsPubKeyHashScriptV0`, and `IsScriptHashScript`.
+3. Determining if a script is a specific type by checking if the relevant
+   version-specific data was extracted.  For example, `ExtractPubKeyHashV0` and
+   `ExtractScriptHashV0`.
+
+The first approach is most suitable for callers that want to work with and
+display information about a wide variety of script types, such as block
+explorers.
+
+The second approach is better suited to callers that are looking for a very
+specific script type as it is more efficient to check for a single type instead
+of checking all types to produce an enum as in the first approach.
+
+The third approach is aimed more at callers that need access to the relevant
+data from the script in addition to merely determining the type.  Unless noted
+otherwise in the documentation for a specific method, the data extraction
+methods only return data if the provided script is actually of the associated
+type.  For example, when a wallet needs to determine if a transaction script is
+a pay-to-pubkey-hash script that involves a particular public key hash, instead
+of calling `IsPubKeyHashScriptV0` (or worse calling `DetermineScriptType` and
+checking the returned enum type) followed by `ExtractPubKeyHashV0`, it is more
+efficient to simply call `ExtractPubKeyHashV0` and check if the data returned is
+not `nil` followed by using said data.
+
+It is also worth noting that the first two approaches offer API methods that
+accept the scripting language version as a parameter as well as their
+version-specific variants that make it clear from the API exactly which script
+types are supported for a given version at compile time.  The third method, on
+the other hand, only offers version-specific methods.  This is discussed further
+the [Extracting Data](#extracting-data) section.
```

Comment:
s / ... further the ... / ... further in the ...

Reply:
Added.